vendor/knative.dev/hack/e2e-tests.sh

```diff
@@ -54,6 +54,7 @@ function go_test_e2e() {
 
 # Setup the test cluster for running the tests.
 function setup_test_cluster() {
+  (
```

Comment:
This might be the culprit. Function is calling user defined functions, and they might define some environment variables, without which tests fail.

Reply:
On https://github.com/knative/serving/pull/13305 in https://github.com/knative/serving/pull/13305/commits/f378d13f87f58f23f104ce922bd6202d083214df I'm testing if removing the subshell helps